### PR TITLE
Update documentation for integer values.

### PR DESCRIPTION
**Pull request by mjdesa**
Documentation for the new integer specification were added.

### CODE REVIEW

**Comment by mjdesa** [comment]
@beckettsean @gunnaraasen @rkuchan Thoughts?

**Comment by gunnaraasen** [comment]
There needs to be a note on those pages that the `i` syntax only applies for 0.9.3 and later. The troubleshooting page should probably include an additional section about the old integer syntax for people running 0.9.0-0.9.2.

**Comment by mjdesa** [comment]
On it.

**Comment by mjdesa** [comment]
@gunnaraasen do you think the note should go on every page that has a reference to writing integers? Or just in one place?

**Comment by beckettsean** [comment]
Only put the notice on the two line protocol pages and the troubleshooting page. Other pages that just reference a write can use the syntax with no version callout

File: content/docs/v0.9/guides/writing_data.md [hunk 1]
```diff
@@ -51,7 +57,7 @@ _Epoch and Precision_
 Timestamps can be supplied as an integer value at the end of the line. The precision is configurable per-request by including a `precision` url parameter. If no precision is specified, the line protocol will default to nanosecond precision. For example to set the time in seconds, use the following request.
 
 ```
-curl -i -XPOST 'http://localhost:8086/write?db=mydb&precision=s' --data-binary 'temperature,machine=unit42,type=assembly external=25,internal=37 1434059627'
+curl -i -XPOST 'http://localhost:8086/write?db=mydb&precision=s' --data-binary 'temperature,machine=unit42,type=assembly external=25i,internal=37i 1434059627'
```
Comment (beckettsean) [inline content/docs/v0.9/guides/writing_data.md]: I don't think we need to change these, unless we specifically reference them as being integers.

File: content/docs/v0.9/introduction/getting_started.md [hunk 2]
```diff
@@ -68,9 +68,9 @@ The following lines are all examples of points that can be written to InfluxDB:
 
 ```
 cpu,host=serverA,region=us_west value=0.64
-payment,device=mobile,product=Notepad,method=credit billed=33,licenses=3 1434067467100293230
-stock,symbol=AAPL bid=127.46,ask=127.48 
-temperature,machine=unit42,type=assembly external=25,internal=37 1434067467000000000
+payment,device=mobile,product=Notepad,method=credit billed=33,licenses=3i 1434067467100293230
+stock,symbol=AAPL bid=127.46,ask=127.48
```
Comment (beckettsean) [inline content/docs/v0.9/introduction/getting_started.md]: I don't think we need to change these, unless we specifically reference them as being integers.

File: content/docs/v0.9/introduction/getting_started.md [hunk 3]
```diff
@@ -102,7 +102,7 @@ _Note: We did not supply a timestamp when writing our point. When no timestamp i
 Let's try storing a different type of data -- sensor data. Enter the following data in the `Values` textbox:
 
 ```sql
-> INSERT temperature,machine=unit42,type=assembly external=25,internal=37
+> INSERT temperature,machine=unit42,type=assembly external=25i,internal=37i
```
Comment (beckettsean) [inline content/docs/v0.9/introduction/getting_started.md]: I don't think we need to change these, unless we specifically reference them as being integers.

File: content/docs/v0.9/write_protocols/line.md [hunk 4]
```diff
@@ -39,8 +39,8 @@ Fields are key-value metrics associated with the measurement.  Every line must h
 
 Field keys are always strings and follow the same syntactical rules as described above for tag keys and values. Field values can be one of four types.  The first value written for a given field on a given measurement defines the type of that field for all series under that measurement.
 
-* _integer_ - Numeric values that do not include a decimal.  (e.g. 1, 345, 2015, -10)
-* _float_ - Numeric values that include a decimal.  (e.g. 1.0, -3.14, 6.0+e5).  Note that all values _must_ have a decimal even if the decimal value is zero (1 is an _integer_, 1.0 is a _float_).
+* _integer_ - Numeric values that do not include a decimal and are followed by a trailing `i` when inserted (e.g. 1i, 345i, 2015i, -10i). Note that all values _must_ have a trailing `i`. If they do not they will be written as floats.
+* _float_ - Numeric values that include a decimal or that are not followed by a trailing `i`.  (e.g. 1, 1.0, -3.14, 6.0+e5, 10).
```
Comment (beckettsean) [inline content/docs/v0.9/write_protocols/line.md]: Shorten to "Numeric values that are not followed by a trailing `i`"

Also, this bulleted list violates the no key-value pairs in bulleted lists in our new style guide. Can you change them to be bold keywords with trailing sentences or something?

File: content/docs/v0.9/write_protocols/line.md [hunk 5]
```diff
@@ -60,7 +64,7 @@ error fatal=true
 event msg="logged out"
 
 # multiple values
-cpu load=10.0,alert=true,reason="value above maximum threshold"
+cpu load=10,alert=true,reason="value above maximum threshold"
```
Comment (beckettsean) [inline content/docs/v0.9/write_protocols/line.md]: Numeric values that include a decimal or that are not followed by a trailing `i`

File: content/docs/v0.9/write_protocols/line.md [hunk 6]
```diff
@@ -73,14 +77,14 @@ an integer epoch in microseconds, milliseconds, seconds, minutes or hours.
...
 A full example is shown below.
 ```
-cpu,host=server01,region=uswest value=1.0 1434055562000000000
-cpu,host=server02,region=uswest value=3.0 1434055562000010000
-temperature,machine=unit42,type=assembly internal=32,external=100 1434055562000000035
-temperature,machine=unit143,type=assembly internal=22,external=130 1434055562005000035
+cpu,host=server01,region=uswest value=1 1434055562000000000
+cpu,host=server02,region=uswest value=3 1434055562000010000
+temperature,machine=unit42,type=assembly internal=32i,external=100i 1434055562000000035
+temperature,machine=unit143,type=assembly internal=22i,external=130i 1434055562005000035
```
Comment (beckettsean) [inline content/docs/v0.9/write_protocols/line.md]: I don't think we need to change these, unless we specifically reference them as being integers.

File: content/docs/v0.9/write_protocols/write_syntax.md [hunk 7]
```diff
@@ -52,46 +52,46 @@ the type of the first point written to given measurement.
 
 ###### Simplest Valid Point (measurement + field)
 ```
-disk_free value=442221834240
+disk_free value=442221834240i
```
Comment (beckettsean) [inline content/docs/v0.9/write_protocols/write_syntax.md]: I don't think we need to change these, unless we specifically reference them as being integers.

File: content/docs/v0.9/write_protocols/write_syntax.md [hunk 8]
```diff
@@ -52,46 +52,46 @@ the type of the first point written to given measurement.
...
 ```
 
 ###### With Timestamp
 ```
-disk_free value=442221834240 1435362189575692182
+disk_free value=442221834240i 1435362189575692182
```
Comment (beckettsean) [inline content/docs/v0.9/write_protocols/write_syntax.md]: I don't think we need to change these, unless we specifically reference them as being integers.

**Comment by mjdesa** [comment]
Made suggested changes.

Left `disk_free` points as integer values.

File: content/docs/v0.9/write_protocols/line.md [hunk 9]
```diff
@@ -39,18 +39,25 @@ Fields are key-value metrics associated with the measurement.  Every line must h
 
 Field keys are always strings and follow the same syntactical rules as described above for tag keys and values. Field values can be one of four types.  The first value written for a given field on a given measurement defines the type of that field for all series under that measurement.
 
-* _integer_ - Numeric values that do not include a decimal.  (e.g. 1, 345, 2015, -10)
-* _float_ - Numeric values that include a decimal.  (e.g. 1.0, -3.14, 6.0+e5).  Note that all values _must_ have a decimal even if the decimal value is zero (1 is an _integer_, 1.0 is a _float_).
-* _boolean_ - A value indicating true or false.  Valid boolean strings are (t, T, true, TRUE, f, F, false, and FALSE).
-* _string_ - A text value.  All string values _must_ be surrounded in double-quotes `"`.  If the string contains
+**Integers** are numeric values that do not include a decimal and are followed by a trailing `i` when inserted (e.g. 1i, 345i, 2015i, -10i). Note that all values _must_ have a trailing `i`. If they do not they will be written as floats.
+
+**Floats** are numeric values that are not followed by a trailing `i`. (e.g. 1, 1.0, -3.14, 6.0+e5, 10).
```
Comment (beckettsean) [inline content/docs/v0.9/write_protocols/line.md]: Scientific notation valid?

File: content/docs/v0.9/write_protocols/write_syntax.md [hunk 10]
```diff
@@ -107,6 +107,8 @@ tag value is `tag,value,with"commas"`, the field key is `field_key\\\\` and the
 
 ## Caveats
 
+> **Note:** Prior to version 0.9.3, integers were numeric values that did not include a decimal (e.g. 1, 345, 2015, -10) and were not followed by a trailing `i`. Including a trailing `i` when writing integers will causes an error in versions 0.9.2 and prior. See [issue](https://github.com/influxdb/influxdb/issues/3519) for more information.
```
Comment (beckettsean) [inline content/docs/v0.9/write_protocols/write_syntax.md]: same "will causes" note

File: content/docs/v0.9/write_protocols/line.md [hunk 11]
```diff
@@ -73,14 +80,16 @@ an integer epoch in microseconds, milliseconds, seconds, minutes or hours.
...
 ```
-cpu,host=server\ 01,region=uswest value=1.0,msg="all systems nominal"
-cpu,host=server\ 01,region=us\,west value_int=1
+cpu,host=server\ 01,region=uswest value=1,msg="all systems nominal"
+cpu,host=server\ 01,region=us\,west value_int=1i
 ```
 In these examples, the "host" is set to `server 01`. The field value associated with field key `msg` is double-quoted, as it is a string. The second example shows a region of `us,west` with the comma properly escaped. In the first example `value` is written as a floating point number. In the second, `value_int` is an integer.
+
+> **Note:** Prior to version 0.9.3, integers were numeric values that did not include a decimal (e.g. 1, 345, 2015, -10) and were not followed by a trailing `i`. Including a trailing `i` when writing integers will causes an error in versions 0.9.2 and prior. See [issue](https://github.com/influxdb/influxdb/issues/3519) for more information.
```
Comment (beckettsean) [inline content/docs/v0.9/write_protocols/line.md]: nit: "will cause" not "will causes"

**Comment by beckettsean** [comment]
A few minor notes, otherwise +1

File: content/docs/v0.9/write_protocols/line.md (hunk 9 above)
Comment (mjdesa) [inline content/docs/v0.9/write_protocols/line.md]: `6.0+e5` is only valid without the `+`. Updating it now

**Comment by mjdesa** [comment]
Closes https://github.com/influxdb/influxdb.com/issues/194

**Pull request merged**